### PR TITLE
#1130 Initial swift package manager support

### PR DESCRIPTION
**Pull request by kenji21**
https://github.com/kif-framework/KIF/issues/1130

As some project refers to openium/master and I don't want to force push to this branch, I created another branch to update my initial PR https://github.com/kif-framework/KIF/pull/1147 to current KIF master

Basically the same changes (removing precompiled headers (.pch) and moving files to be "more" SPM compliant)

### CODE REVIEW

**Comment by kenji21** [comment]
Since previous PR, Github Actions have been added, waiting for it to run

I still need to cherry-pick my latest commit demoing integration using "path package" of KIF in a test project and ask review to @dostrander or @justinseanmartin

**Comment by kenji21** [comment]
Got Xcode 10.3 back... got issue on AutoCorrectTests::testSmartQuotesEnabled, and on AutoCorrectTests_ViewTestActor::testSmartQuotesEnabled (2 assertions per test => 4 failures), but not on the ones of Github Actions... 🙁

```
Test Suite 'KIF Tests - XCTest.xctest' failed at 2020-11-13 09:23:58.324.
	 Executed 266 tests, with 4 failures (4 unexpected) in 872.233 (880.675) seconds
Test Suite 'All tests' failed at 2020-11-13 09:23:58.325.
	 Executed 266 tests, with 4 failures (4 unexpected) in 872.233 (880.677) seconds
```

**Comment by kenji21** [comment]
Same result with ci script :
```
export DEVELOPER_DIR=/Applications/Xcode_10.3.app/Contents/Developer
./scripts/ci.sh 
```

```
Executed 266 tests, with 4 failures (4 unexpected) in 882.941 (890.868) seconds
```

**Comment by kenji21** [comment]
tried again by running ci script in a ssh session... same errors

```
	 Executed 266 tests, with 4 failures (4 unexpected) in 884.942 (893.389) seconds

2020-11-13 10:42:43.565 xcodebuild[25708:59884020] [MT] IDETestOperationsObserverDebug: 904.832 elapsed -- Testing started completed.
2020-11-13 10:42:43.565 xcodebuild[25708:59884020] [MT] IDETestOperationsObserverDebug: 0.000 sec, +0.000 sec -- start
2020-11-13 10:42:43.565 xcodebuild[25708:59884020] [MT] IDETestOperationsObserverDebug: 904.832 sec, +904.832 sec -- end

Failing tests:
	KIF Tests:
		-[AutocorrectTests testSmartQuotesEnabled]
		-[AutocorrectTests testSmartQuotesEnabled]
		-[AutocorrectTests_ViewTestActor testSmartQuotesEnabled]
		-[AutocorrectTests_ViewTestActor testSmartQuotesEnabled]
```

I suspect than user interaction is not enabled, so all tests are failing (maybe because of user not logged in on CI server)

**Comment by kenji21** [comment]
OK, tried Actions on another project, it is not "user interaction disabled" issue

As Xcode run tests alphabetically, I move the [BackgroundTests testBackgroundApp] to the end by renaming its class to ZBackgroundTests:

```
ZBackgroundTests

    ✓ testBackgroundApp (3.880 seconds)

TableViewTests
  testTappingLastRowAndSectionWithoutAnimation, The step timed out after 10.00 seconds: Found an accessibility element with the label "Last Cell", but not with the traits "8"
  /Users/runner/work/KIF/KIF/KIF Tests/TableViewTests.m:57
    [tester tapRowAtIndexPath:[NSIndexPath indexPathForRow:-1 inSection:-1] inTableViewWithAccessibilityIdentifier:@"TableView Tests Table"];
    [tester waitForViewWithAccessibilityLabel:@"Last Cell" traits:UIAccessibilityTraitSelected];
}
  testTappingLastRowAndSectionWithoutAnimation, failed: caught "KIFFailureException", "The step timed out after 10.00 seconds: Found an accessibility element with the label "Last Cell", but not with the traits "8""
  <unknown>:0

	 Executed 266 tests, with 2 failures (2 unexpected) in 977.533 (986.589) seconds
```

**Comment by kenji21** [comment]
Force pushed to trigger another build because first time iPad build was stopped because of fail in iPhone tests
So it works on iPad, but not on iPhone X, 🤔
let's try using an iPhone 8 instead of X

**Comment by kenji21** [comment]
so there is an issue with `testTappingLastRowAndSectionWithoutAnimation` but only on `iPhone X` maybe due to safe areas

**Comment by kenji21** [comment]
nice, CI works for building SPM package, here is the command added to `scripts/ci.sh` to build the package for the simulator:

```
swift build -Xcc "-isysroot" -Xcc "$(xcrun --sdk iphonesimulator --show-sdk-path)" -Xcc "-target" -Xcc "x86_64-apple-ios$(xcrun --sdk iphonesimulator --show-sdk-version)-simulator"
```

**Comment by kenji21** [comment]
As there are 3 projects built by CI (`Testable Swift.xcodeproj`, `Testable.xcodeproj` and `Calculator.xcodeproj`), to not duplicate code, @justinseanmartin @dostrander what do you think of modifying one to integrate the framework using SPM (doing so will require Xcode 11 minimum for this project, thus adding Xcode 11 in the CI matrix)

**Comment by justinseanmartin** [comment]
Seems like a reasonable plan to me, but defer to @dostrander.

On Nov 14, 2020, at 8:27 AM, kenji21 <notifications@github.com> wrote:

﻿

As there are 3 projects built by CI (Testable Swift.xcodeproj, Testable.xcodeproj and Calculator.xcodeproj), to not duplicate code, @justinseanmartin<https://github.com/justinseanmartin> @dostrander<https://github.com/dostrander> what do you think of modifying one to integrate the framework using SPM (doing so will require Xcode 11 minimum for this project, thus adding Xcode 11 in the CI matrix)

**Comment by dostrander** [comment]
@kenji21 Sounds good to me

**Comment by kenji21** [comment]
Ok, as `Calculator.xcodeproj` uses `KIF.framework`, `Testable.xcodeproj`uses KIF static library and `Testable Swift.xcodeproj` also uses `KIF.framework`, I have removed KIF.xcodeproj and drag & dropped KIF folder (Xcode find it is a Swift Package) in `Testable Swift.xcodeproj` :

Screenshot 2

**Comment by kenji21** [comment]
Tested on my mac using `export DEVELOPER_DIR="/Applications/Xcode_11.7.app/Contents/Developer"` and then running

On `Xcode 11.7` with an `iPhone SE (2nd generation)` many tests fails (https://github.com/kif-framework/KIF/pull/1193/checks?check_run_id=1408774849):
```
 Executed 266 tests, with 474 failures (474 unexpected) in 2923.099 (2933.903) seconds
```

Found one issue:

```objc
-(void) testPullToRefreshByAccessibilityIdentifierWithDuration
{
	UITableView *tableView;
	[tester waitForAccessibilityElement:NULL view:&tableView withIdentifier:@"Test Suite TableView" tappable:NO];

    [tester tapViewWithAccessibilityLabel:@"Reset Refresh Control"];
	[tester pullToRefreshViewWithAccessibilityIdentifier:@"Test Suite TableView" pullDownDuration:KIFPullToRefreshInAboutThreeSeconds];
	[tester waitForViewWithAccessibilityLabel:@"Bingo!"];
	[tester waitForAbsenceOfViewWithAccessibilityLabel:@"Bingo!"];
}
```

makes refresh control label to have text "Refreshing...", and then `waitForViewWithAccessibilityLabel:"Bingo!"` fails

```objc
- (IBAction)resetRefreshControl
{
	self.refreshControl.attributedTitle = [[NSAttributedString alloc] initWithString:NSLocalizedString(@"Refreshing...", @"") attributes:nil];
}
```

To fix this:

```diff
 -(void)pullToRefreshHandler
 {
-       self.refreshControl.attributedTitle = [[NSAttributedString alloc] initWithString:NSLocalizedString(@"Bingo!", @"") attributes:nil];
     dispatch_after(dispatch_time(DISPATCH_TIME_NOW, (int64_t)(4.0 * NSEC_PER_SEC)), dispatch_get_main_queue(), ^{
+        self.refreshControl.attributedTitle = [[NSAttributedString alloc] initWithString:NSLocalizedString(@"Bingo!", @"") attributes:nil];
         [self.refreshControl endRefreshing];
         [self.tableView setContentOffset:CGPointZero animated:YES];
     });
```

Found another one that could be the source of many issues: the photo selector is shown and not cancelled, so I moved SystemAlertTests to the end of execution

Next actions should give a result like: `Executed 266 tests, with 34 failures (34 unexpected) in 993.439 (1001.637) seconds`

**Comment by kenji21** [comment]
Run [1410490218](https://github.com/kif-framework/KIF/pull/1193/checks?check_run_id=1410490218) gets better than expected: 
```
	 Executed 266 tests, with 24 failures (24 unexpected) in 973.740 (982.695) seconds

Failing tests:
	KIF Tests:
		-[AutocorrectTests testSmartQuotesEnabled]
		-[AutocorrectTests testSmartQuotesEnabled]
		-[AutocorrectTests_ViewTestActor testSmartQuotesEnabled]
		-[AutocorrectTests_ViewTestActor testSmartQuotesEnabled]
		-[TypingTests testEnteringTextIntoFirstResponder]
		-[TypingTests testEnteringTextIntoFirstResponder]
		-[TypingTests testEnteringTextIntoViewWithAccessibilityLabel]
		-[TypingTests testEnteringTextIntoViewWithAccessibilityLabel]
		-[TypingTests_ViewTestActor testEnteringTextIntoFirstResponder]
		-[TypingTests_ViewTestActor testEnteringTextIntoFirstResponder]
		-[TypingTests_ViewTestActor testEnteringTextIntoViewWithAccessibilityLabel]
		-[TypingTests_ViewTestActor testEnteringTextIntoViewWithAccessibilityLabel]
		-[ZSystemAlertTests testAuthorizingPhotosAccess]
		-[ZSystemAlertTests testAuthorizingPhotosAccess]
		-[ZSystemAlertTests testAuthorizingPhotosAccess]
		-[ZSystemAlertTests testAuthorizingPhotosAccess]
		-[ZSystemAlertTests_ViewTestActor testAuthorizingLocationServices]
		-[ZSystemAlertTests_ViewTestActor testAuthorizingLocationServices]
		-[ZSystemAlertTests_ViewTestActor testAuthorizingLocationServices]
		-[ZSystemAlertTests_ViewTestActor testAuthorizingLocationServices]
		-[ZSystemAlertTests_ViewTestActor testAuthorizingPhotosAccess]
		-[ZSystemAlertTests_ViewTestActor testAuthorizingPhotosAccess]
		-[ZSystemAlertTests_ViewTestActor testAuthorizingPhotosAccess]
		-[ZSystemAlertTests_ViewTestActor testAuthorizingPhotosAccess]


** TEST FAILED **
```

As expected, System Alerts Tests are broken

**Comment by kenji21** [comment]
After fixing typing tests:
```
Executed 266 tests, with 16 failures (16 unexpected) in 928.241 (936.347) seconds
```

**Comment by kenji21** [comment]
Smart dashes fixed:
```
	 Executed 266 tests, with 16 failures (16 unexpected) in 928.241 (936.347) seconds

Failing tests:
	KIF Tests:
		-[AutocorrectTests testSmartQuotesEnabled]
		-[AutocorrectTests testSmartQuotesEnabled]
		-[AutocorrectTests_ViewTestActor testSmartQuotesEnabled]
		-[AutocorrectTests_ViewTestActor testSmartQuotesEnabled]
		-[ZSystemAlertTests testAuthorizingPhotosAccess]
		-[ZSystemAlertTests testAuthorizingPhotosAccess]
		-[ZSystemAlertTests testAuthorizingPhotosAccess]
		-[ZSystemAlertTests testAuthorizingPhotosAccess]
		-[ZSystemAlertTests_ViewTestActor testAuthorizingLocationServices]
		-[ZSystemAlertTests_ViewTestActor testAuthorizingLocationServices]
		-[ZSystemAlertTests_ViewTestActor testAuthorizingLocationServices]
		-[ZSystemAlertTests_ViewTestActor testAuthorizingLocationServices]
		-[ZSystemAlertTests_ViewTestActor testAuthorizingPhotosAccess]
		-[ZSystemAlertTests_ViewTestActor testAuthorizingPhotosAccess]
		-[ZSystemAlertTests_ViewTestActor testAuthorizingPhotosAccess]
		-[ZSystemAlertTests_ViewTestActor testAuthorizingPhotosAccess]
```

let's continue by loading UIAutomation.framework correctly (note that UIAutomation.framework no longer exists within Xcode 12)

**Comment by kenji21** [comment]
Nice,

Fixed AutocorrectTests, now 16 -> 12 fails, now time to fix System Alert Tests:

```
 Executed 266 tests, with 12 failures (12 unexpected) in 914.926 (923.222) seconds

Failing tests:
	KIF Tests:
		-[ZSystemAlertTests testAuthorizingPhotosAccess]
		-[ZSystemAlertTests testAuthorizingPhotosAccess]
		-[ZSystemAlertTests testAuthorizingPhotosAccess]
		-[ZSystemAlertTests testAuthorizingPhotosAccess]
		-[ZSystemAlertTests_ViewTestActor testAuthorizingLocationServices]
		-[ZSystemAlertTests_ViewTestActor testAuthorizingLocationServices]
		-[ZSystemAlertTests_ViewTestActor testAuthorizingLocationServices]
		-[ZSystemAlertTests_ViewTestActor testAuthorizingLocationServices]
		-[ZSystemAlertTests_ViewTestActor testAuthorizingPhotosAccess]
		-[ZSystemAlertTests_ViewTestActor testAuthorizingPhotosAccess]
		-[ZSystemAlertTests_ViewTestActor testAuthorizingPhotosAccess]
		-[ZSystemAlertTests_ViewTestActor testAuthorizingPhotosAccess]
```

**Comment by kenji21** [comment]
Ok, I was looking to `xcrun simctl privacy help` to deprecate the acknowledgeSystemAlert... it's ok for Location, but it doesn't provide support for Bluetooth / Push notifications ones...

**Comment by kenji21** [comment]
As this PR was only intended to add SPM support, can we merge it, and treat the `Xcode 11` acknowledgeSystemAlert issue support on another PR ?
I can squash all my commit into one if needed
Another issue is supporting devices having "safe areas" (e.g. iPhone X) as I saw that test were failing because of this
Another issue will be `Xcode 12` as UIAutomation framework is not more present (test will fail see https://github.com/openium/KIF/commit/ee59ab8d8b87b5550350e2a5b40a414a86474e80#diff-e6c4dd745e29178b755895d5d06f10181fe788142825897ca4d6baed20e7f84dL216)

**Comment by dostrander** [comment]
@kenji21 Can we add an early return if we detect that it is running xcode 11 for the system tests

**Comment by kenji21** [comment]
> @kenji21 Can we add an early return if we detect that it is running xcode 11 for the system tests

sure, i'll have to rebase my branch on master (pbxproj have many conflicts)

**Comment by kenji21** [comment]
> As this PR was only intended to add SPM support, can we merge it, and treat the `Xcode 11` acknowledgeSystemAlert issue support on another PR ?

=> https://github.com/kif-framework/KIF/issues/1156

> I can squash all my commit into one if needed

> Another issue is supporting devices having "safe areas" (e.g. iPhone X) as I saw that test were failing because of this

=> https://github.com/kif-framework/KIF/issues/1196

> Another issue will be `Xcode 12` as UIAutomation framework is not more present (test will fail see [openium@ee59ab8#diff-e6c4dd745e29178b755895d5d06f10181fe788142825897ca4d6baed20e7f84dL216](https://github.com/openium/KIF/commit/ee59ab8d8b87b5550350e2a5b40a414a86474e80#diff-e6c4dd745e29178b755895d5d06f10181fe788142825897ca4d6baed20e7f84dL216))

=> Fixed by early return

So Xcode 11 tests are OK, but carthage build using Xcode 12 is failing

**Comment by kenji21** [comment]
So the error is that `xcodebuild archive` makes a slice for Apple Silicon for the simulator:

```
kenji@macboo:~/github/KIF [spm-support]
file /Users/kenji/Library/Developer/Xcode/DerivedData/Testable-aotjmdlwrwsqaybiicbddpsjjgjp/Build/Intermediates.noindex/ArchiveIntermediates/KIFFramework/IntermediateBuildFilesPath/UninstalledProducts/iphoneos/KIF.framework/KIF
[....]/UninstalledProducts/iphoneos/KIF.framework/KIF: Mach-O universal binary with 2 architectures: [arm_v7:Mach-O dynamically linked shared library arm_v7] [arm64:Mach-O 64-bit dynamically linked shared library arm64]
[....]/UninstalledProducts/iphoneos/KIF.framework/KIF (for architecture armv7):	Mach-O dynamically linked shared library arm_v7
[....]/UninstalledProducts/iphoneos/KIF.framework/KIF (for architecture arm64):	Mach-O 64-bit dynamically linked shared library arm64
```

```
kenji@macboo:~/github/KIF [spm-support]
file /Users/kenji/Library/Developer/Xcode/DerivedData/Testable-aotjmdlwrwsqaybiicbddpsjjgjp/Build/Products/Release-iphonesimulator/KIF.framework/KIF
[....]/Products/Release-iphonesimulator/KIF.framework/KIF: Mach-O universal binary with 3 architectures: [x86_64:Mach-O 64-bit dynamically linked shared library x86_64] [i386:Mach-O dynamically linked shared library i386] [arm64:Mach-O 64-bit dynamically linked shared library arm64]
[....]/Products/Release-iphonesimulator/KIF.framework/KIF (for architecture x86_64):	Mach-O 64-bit dynamically linked shared library x86_64
[....]/Products/Release-iphonesimulator/KIF.framework/KIF (for architecture i386):	Mach-O dynamically linked shared library i386
[....]/Products/Release-iphonesimulator/KIF.framework/KIF (for architecture arm64):	Mach-O 64-bit dynamically linked shared library arm64
```

```
kenji@macboo:~/github/KIF [spm-support]
fatal error:[....]/UninstalledProducts/iphoneos/KIF.framework/KIF and [....]/Build/Products/Release-iphonesimulator/KIF.framework/KIF have the same architectures (arm64) and can't be in the same fat output file
```

Related to https://github.com/Carthage/Carthage/issues/3019

**Comment by kenji21** [comment]
maybe travis CI can be removed now

**Comment by dostrander** [comment]
What is the xcode 12 failure from carthage?

The reason we use travis still is because iOS 11 is not supported currently on github actions and KIF still supports it.

**Comment by kenji21** [comment]
Sorry, I haven't see it (confused by Xcode/iOS versions), it seems that's always the System Alert Tests that are also failing for iOS 11. :/

```
Failing tests:
3178	KIF Tests:
3179		-[ZSystemAlertTests testAuthorizingLocationServicesAndNotificationsScheduling]
3180		-[ZSystemAlertTests testAuthorizingLocationServicesAndNotificationsScheduling]
3181		-[ZSystemAlertTests testAuthorizingLocationServicesAndNotificationsScheduling]
3182		-[ZSystemAlertTests testAuthorizingLocationServicesAndNotificationsScheduling]
3183		-[ZSystemAlertTests testAuthorizingPhotosAccess]
3184		-[ZSystemAlertTests testAuthorizingPhotosAccess]
3185		-[ZSystemAlertTests testAuthorizingPhotosAccess]
3186		-[ZSystemAlertTests testAuthorizingPhotosAccess]
3187		-[ZSystemAlertTests_ViewTestActor testAuthorizingLocationServices]
3188		-[ZSystemAlertTests_ViewTestActor testAuthorizingLocationServices]
3189		-[ZSystemAlertTests_ViewTestActor testAuthorizingLocationServices]
3190		-[ZSystemAlertTests_ViewTestActor testAuthorizingLocationServices]
3191		-[ZSystemAlertTests_ViewTestActor testAuthorizingPhotosAccess]
3192		-[ZSystemAlertTests_ViewTestActor testAuthorizingPhotosAccess]
3193		-[ZSystemAlertTests_ViewTestActor testAuthorizingPhotosAccess]
3194		-[ZSystemAlertTests_ViewTestActor testAuthorizingPhotosAccess]
```

Just rebased on top of current master (this one was easier than the previous one, merging a pbxproj is hard)

**Comment by dostrander** [comment]
@kenji21 they also use the UIAutomation framework

**Comment by kenji21** [comment]
I am missing something, executed them on our CI after installing Xcode 11.6 and downloading simulator 11.4 (as it is in [.travis.yml](https://github.com/kif-framework/KIF/blob/master/.travis.yml#L4) from current master)

```shell
$ export DEVELOPER_DIR=/Applications/Xcode_11.6.app/Contents/Developer/
$ SIMULATOR="name=iPad Pro (12.9-inch) (2nd generation),OS=11.4"
$ ./scripts/ci.sh "${SIMULATOR}"
[...]
ZBackgroundTests
    ✓ testBackgroundApp (13.208 seconds)
ZSystemAlertTests
    ✓ testAuthorizingLocationServicesAndNotificationsScheduling (1.540 seconds)
    ✓ testAuthorizingPhotosAccess (1.538 seconds)
ZSystemAlertTests_ViewTestActor
    ✓ testAuthorizingLocationServices (1.539 seconds)
    ✓ testAuthorizingPhotosAccess (1.538 seconds)


	 Executed 275 tests, with 0 failures (0 unexpected) in 759.207 (768.921) seconds
[...]
```

what surprises me is that Xcode 11.6 ship with iOS 13.6 see [xcodereleases.com](https://xcodereleases.com/alpha.html)

**Comment by kenji21** [comment]
ok, it breaks in the `beforeEach()` method

```
ZSystemAlertTests
  testAuthorizingLocationServicesAndNotificationsScheduling, The step timed out after 10.00 seconds: Failed to find accessibility element with the label "System Alerts"
  /Users/travis/build/kif-framework/KIF/KIF Tests/SystemAlertTests.m:30
{
    [tester tapViewWithAccessibilityLabel:@"System Alerts"];
}

```

**Comment by kenji21** [comment]
It seems that one test is flaky...
```
Failing tests:
	KIF Tests:
		-[WebViewTests_ViewTestActor testTappingLinks]

** TEST FAILED **
```

while my previous push was fine:

Screenshot 11

File: .github/workflows/build.yml
Comment (dostrander) [inline .github/workflows/build.yml:16]: nit: remove comments

File: KIF Tests/BackgroundTests.m
Comment (dostrander) [inline KIF Tests/BackgroundTests.m:11]: rename back to BackgroundTests

File: KIF Tests/BackgroundTests.m
Comment (dostrander) [inline KIF Tests/BackgroundTests.m:38]: could we just do `if (@available(iOS 14.0, *))`  ??

File: KIF Tests/SystemAlertTests.m
Comment (dostrander) [inline KIF Tests/SystemAlertTests.m:11]: Rename back to `SystemAlertTests`

File: KIF Tests/SystemAlertTests.m
Comment (dostrander) [inline KIF Tests/SystemAlertTests.m:24]: Can we use @available instead of `#if`s ?

File: KIF Tests/SystemAlertTests_ViewTestActor.m
Comment (dostrander) [inline KIF Tests/SystemAlertTests_ViewTestActor.m:12]: Change back to `SystemAlertTests_ViewTestACtor`

File: Sources/KIF/KIFAccessibilityEnabler.m
Comment (dostrander) [inline Sources/KIF/KIFAccessibilityEnabler.m:23]: why did this change?

File: Sources/KIF/UIAutomationHelper.m
Comment (dostrander) [inline Sources/KIF/UIAutomationHelper.m:217]: Does this help anything for uiautomation?

File: .github/workflows/build.yml
Comment (kenji21) [reply .github/workflows/build.yml]: done in https://github.com/openium/KIF/commit/4f286cf6f0264c1b93b5b1f860f00ca0bfb43104

File: KIF Tests/BackgroundTests.m
Comment (kenji21) [reply KIF Tests/BackgroundTests.m]: done in https://github.com/openium/KIF/commit/4f286cf6f0264c1b93b5b1f860f00ca0bfb43104

File: KIF Tests/BackgroundTests.m
Comment (kenji21) [reply KIF Tests/BackgroundTests.m]: done in https://github.com/openium/KIF/commit/4f286cf6f0264c1b93b5b1f860f00ca0bfb43104

File: KIF Tests/SystemAlertTests.m
Comment (kenji21) [reply KIF Tests/SystemAlertTests.m]: done in https://github.com/openium/KIF/commit/4f286cf6f0264c1b93b5b1f860f00ca0bfb43104

File: KIF Tests/SystemAlertTests.m
Comment (kenji21) [reply KIF Tests/SystemAlertTests.m]: done in https://github.com/openium/KIF/commit/4f286cf6f0264c1b93b5b1f860f00ca0bfb43104

File: KIF Tests/SystemAlertTests_ViewTestActor.m
Comment (kenji21) [reply KIF Tests/SystemAlertTests_ViewTestActor.m:12]: done in https://github.com/openium/KIF/commit/4f286cf6f0264c1b93b5b1f860f00ca0bfb43104

File: Sources/KIF/KIFAccessibilityEnabler.m
Comment (kenji21) [reply Sources/KIF/KIFAccessibilityEnabler.m]: to use it from UIAutomationHelper.m, reverted in https://github.com/openium/KIF/commit/4f286cf6f0264c1b93b5b1f860f00ca0bfb43104

File: Sources/KIF/UIAutomationHelper.m
Comment (kenji21) [reply Sources/KIF/UIAutomationHelper.m]: tried to find why UIAutomation fails to load, see that `loadDylibForSimulator` prepends IPHONE_SIMULATOR_ROOT to framework path, but it didn't helped, reverted in https://github.com/openium/KIF/commit/4f286cf6f0264c1b93b5b1f860f00ca0bfb43104

**Review by dostrander — APPROVED** [review]
Thanks for doing this, i know a bunch of folks have been really excited for this

**Comment by kikeenrique** [comment]
Can I ask for the plans to merge this?
I'm one of those folks excited for having SPM integration.

**Comment by kikeenrique** [comment]
FYI, I've tested https://github.com/openium/KIF/tree/spm-support and found no problems so far in my projects.
Looks quite promising! Nice work! (hope it gets merged soon 🤞)

**Comment by kenji21** [comment]
> FYI, I've tested https://github.com/openium/KIF/tree/spm-support and found no problems so far in my projects.
> Looks quite promising! Nice work! (hope it gets merged soon 🤞)

please use https://github.com/openium/KIF/tree/spm-support-freez as I may rewrite history (if squash needed for example) and force push on spm-support branch

**Comment by kikeenrique** [comment]
> > FYI, I've tested https://github.com/openium/KIF/tree/spm-support and found no problems so far in my projects.
> 
> > Looks quite promising! Nice work! (hope it gets merged soon 🤞)
> 
> 
> 
> please use https://github.com/openium/KIF/tree/spm-support-freez as I may rewrite history (if squash needed for example) and force push on spm-support branch

Thanks for the info 😀.
Nevertheless, probably, after having tested it, I will patiently wait for an official release.

**Comment by dostrander** [comment]
@kenji21  This looks good to me once you are ready to merge  let me know. Only reason i'm not currently is it still has the WIP title :)

**Comment by dostrander** [comment]
@kenji21 This has duplicate files, moved the files around without fixing the podspec, and not including anything in the IdentifierTests subspec. I need to revert this for now. If you could fix these issues that would be helpful. I am going to add a check to make sure that the podspec should validate which would have picked up these issues.

**Pull request merged**
